src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { getUserById } from '../../servers/userServer';
+import { getNewTodoId } from '../../servers/todoServer';
+import usersFromServer from '../../api/users';
+
+type Props = {
+  todos: Todo[];
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ todos, onSubmit }) => {
+  //#region states
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  //#endregion
+  //#region change handlers
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+  //#endregion
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isValidTitle = /^[a-zA-Z\d\s]+$/.test(title.trim());
+
+    setHasTitleError(!isValidTitle);
+    setHasUserIdError(!userId);
+
+    if (!isValidTitle || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: getNewTodoId(todos),
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
```

Comment:
The use of `getUserById` here violates the task requirement to centralize data preparation in the `App` component. Consider moving this logic to the `App` component and passing the necessary user data as props to `TodoForm`.